Windows/Debugger/Debugger_MemoryDlg.cpp

```diff
@@ -84,107 +96,130 @@ void CMemoryDlg::Update(void)
 {
 	if (m_hDlg != NULL)
 	{
-		CtrlMemView *mv = CtrlMemView::getFrom(GetDlgItem(m_hDlg,IDC_MEMVIEW));
-		if (mv != NULL)
-			mv->redraw();
+		if (memView != NULL)
+			memView->redraw();
 	}	
 }
 
+void CMemoryDlg::searchBoxRedraw(std::vector<u32> results) {
+
+	int index;
+	wchar_t temp[256];
+	//std::lock_guard<std::recursive_mutex> guard(lock_);
+	SendMessage(srcListHdl, WM_SETREDRAW, FALSE, 0);
+	ListBox_ResetContent(srcListHdl);
+	for (int i = 0; i < results.size(); i++) {
+		wsprintf(temp, L"0x%08X", results[i]);
+		index = (int)ListBox_AddString(srcListHdl,temp);
+		ListBox_SetItemData(srcListHdl, index, results[i]);
+	}
+   	SendMessage(srcListHdl, WM_SETREDRAW, TRUE, 0);
+	RedrawWindow(srcListHdl, NULL, NULL, RDW_ERASE | RDW_FRAME | RDW_INVALIDATE | RDW_ALLCHILDREN);
+}
+
+
 void CMemoryDlg::NotifyMapLoaded()
 {
 	if (m_hDlg)
 	{
-		HWND list = GetDlgItem(m_hDlg,IDC_SYMBOLS);
+
 		if (g_symbolMap)
-			g_symbolMap->FillSymbolListBox(list, ST_DATA);
-		HWND lb = GetDlgItem(m_hDlg,IDC_REGIONS);
-		int sel = ComboBox_GetCurSel(lb);
-		ComboBox_ResetContent(lb);
+			g_symbolMap->FillSymbolListBox(symListHdl, ST_DATA);
+		int sel = ComboBox_GetCurSel(memViewHdl);
+		ComboBox_ResetContent(memViewHdl);
+
     /*
 		for (int i = 0; i < cpu->getMemMap()->numRegions; i++)
 		{
 			// TODO: wchar_t
 			int n = ComboBox_AddString(lb,cpu->getMemMap()->regions[i].name);
 			ComboBox_SetItemData(lb,n,cpu->getMemMap()->regions[i].start);
 		}*/
-		ComboBox_SetCurSel(lb,sel>=0?sel:0);
+		ComboBox_SetCurSel(memViewHdl,sel>=0?sel:0);
 	}
 	Update(); 
 }
 
-
 BOOL CMemoryDlg::DlgProc(UINT message, WPARAM wParam, LPARAM lParam)
 {
-
-	switch(message)
-	{
-	case WM_COMMAND:
-		{
-			CtrlMemView *mv = CtrlMemView::getFrom(GetDlgItem(m_hDlg,IDC_MEMVIEW));
-			switch (LOWORD(wParam))
-			{
-			case IDC_REGIONS:
-				switch (HIWORD(wParam)) 
-				{ 
-				case LBN_DBLCLK:
-					{
-						HWND lb = GetDlgItem(m_hDlg,LOWORD(wParam));
-						int n = ComboBox_GetCurSel(lb);
-						if (n!=-1)
-						{
-							unsigned int addr = (unsigned int)ComboBox_GetItemData(lb,n);
-							mv->gotoAddr(addr);
+	switch(message){
+		case WM_COMMAND:{
```

Comment:
nit: Prefer case at the same indent as switch.  There's inconsistent indentation below now since some of the first ones are indented.

-[Unknown]

Reply:
yes, that was sloppy of me